app/assets/javascripts/discourse/app/routes/discovery.js

```diff
@@ -46,25 +46,21 @@ export default DiscourseRoute.extend(OpenComposer, {
 
   actions: {
     loading() {
-      this.controllerFor("discovery").set("loading", true);
-      return true;
+      this.controllerFor("discovery").loadingBegan();
     },
 
     loadingComplete() {
-      this.controllerFor("discovery").set("loading", false);
+      this.controllerFor("discovery").loadingComplete();
       if (!this.session.get("topicListScrollPosition")) {
         scrollTop();
       }
-      return false;
```

Comment:
These `return true` things are not necessary in actions. I'm not sure they ever were. Probably wishful thinking!